### DISCUSSION TITLE
Access to path denied during artifact upload

### DISCUSSION THREAD

**Discussion by Synchro** [community/community#25875] [Actions]
<p>I have <a href="https://github.com/PHPMailer/PHPMailer/blob/master/.github/workflows/docs.yaml" rel="nofollow noopener">a docs build action</a> for PHPMailer that passes artifacts between two jobs in a single workflow, but it’s <a href="https://github.com/PHPMailer/PHPMailer/commit/3563788a8620ee003b255eccbc3d75d20ef92d99/checks?check_suite_id=247931531" rel="nofollow noopener">failing with this error</a>:</p>
<pre><code>Uploading artifact 'docs' from '/home/runner/work/PHPMailer/PHPMailer/docs' for run #3
##[error]Access to the path '/home/runner/work/PHPMailer/PHPMailer/docs/phpdoc-cache-8b' is denied.
##[error]Exit code 1 returned from process: file name '/home/runner/runners/2.158.0/bin/Runner.PluginHost', arguments 'action "GitHub.Runner.Plugins.Artifact.PublishArtifact, Runner.Plugins"'.
</code></pre>
<p>As far as I can see I’ve followed what <a href="https://help.github.com/en/articles/persisting-workflow-data-using-artifacts#passing-data-between-jobs-in-a-workflow" rel="nofollow noopener"> the docs</a> say - so why don’t I have access to this path?</p>

**Comment by chrispat** [comment]
<p>In the case of actions that use a Dockerfile those files are probably created as the root user inside the container.  Because the upload-artifact action runs as the runner user you will need to sudo chmod the permissions on those files so the runner user can read them.</p>

**Comment by Synchro** [comment, marked as answer]
<p>I attempted to chmod the path that it was complaining about, but it said it didn’t exist!</p>
<p>A solution was found by merging the build and publish steps into one job, so that it didn’t need to use the upload/download steps at all.</p>
